core/src/main/scala/cats/instances/vector.scala

```diff
@@ -90,9 +90,6 @@ trait VectorInstances extends cats.kernel.instances.VectorInstances {
 
       override def collect[A, B](fa: Vector[A])(f: PartialFunction[A, B]): Vector[B] = fa.collect(f)
 
-      override def foldM[G[_], A, B](fa: Vector[A], z: B)(f: (B, A) => G[B])(implicit G: Monad[G]): G[B] =
-        Foldable.iteratorFoldM(fa.toIterator, z)(f)
```

Comment:
`Vector.tail` is so slow, we are probably better off doing `fa.toList.foldM` here if we do the above.